src/utils/lstring.ts

```diff
@@ -29,14 +31,30 @@ export default function (ltext: string | LString | { [locale: string]: string }
   return text;
 }
 
-export function trim(ltext: LString) {
+export function trim(ltext?: string | LString | { [locale: string]: string }) {
+  if (!ltext) return ltext;
+
+  if (typeof (ltext) === 'string') return ltext.trim();
+
   return Object.entries(ltext).reduce((ret, [k, v]) => ({ ...ret, [k]: v?.trim() }), {});
 }
 
-export function isNullOrEmpty(ltext?: LString) {
-  return !ltext || !Object.values(ltext).some(Boolean);
+export function isNullOrEmpty(ltext?: string | LString | { [locale: string]: string }) {
+  return !ltext || !Object.values(cleanLString(ltext)).some(Boolean);
+}
+
+export function isNullOrWhiteSpace(ltext?: string | LString | { [locale: string]: string }) {
+  return !ltext || !Object.values(trim(cleanLString(ltext))).some(Boolean);
 }
 
-export function isNullOrWhiteSpace(ltext?: LString) {
-  return !ltext || !Object.values(trim(ltext)).some(Boolean);
+function isValidLocaleEntry([k, v]: any) {
+  return k && !k.startsWith("#")
 }
+
+function cleanLString(ltext?: string | LString | { [locale: string]: string }): string | LString | { [locale: string]: string } | null | undefined {
+  if (!ltext || typeof (ltext) === 'string') return ltext;
```

Comment:
don't think it has to accept/return anything else than lstring 
```suggestion
function cleanLString(ltext?: LString | { [locale: string]: string }): LString | { [locale: string]: string } {
```

Reply:
@sbilodeau should we be strict about it and not even check for `typeof (ltext) === 'string')`?

Reply:
yes my bad. i meant to remove the line. I fixed the suggestion